### ISSUE TITLE
Error adding access list. (413)

### ISSUE THREAD

**Issue by adriadam10** [lavinir/npm-geoaccesslists#1]
I get this error when is adding the access list

**Comment by lavinir** [comment]
Yeah, that's a request too large error.. I guess it went over the limit allowed in the NPM admin API. 
I'll try to open an issue there see if there's a way to expand it. 
How large is your IP file ?

**Comment by yuriitsarenko** [comment]
I have the same problem, no solution yet?

**Comment by cfrans** [comment]
Same error here, 27mb file.
Maybe there is a way to split the rules based on file size and create more than one rule?

**Comment by cfrans** [comment]
> Splitting the files would be great if npm allow to use more than one access rule per domain

Ah, I did not knew that ;(

**Comment by lavinir** [comment]
Yes unfortunately, this is kind of a limitation of the API npm allows at the moment..

**Comment by mxm199** [comment]
Good afternoon, have you found a solution to this problem?

Judging by the description, this error can be corrected by the client_max_body_size field in nginx.conf, but I have it set to 2000m...

**Comment by diegocade1** [comment]
any update on this?

**Comment by mxm199** [comment]
> any update on this?

No

**Comment by bitKrakenCode** [comment]
I did implement a workaround for me, by modifying the nginxproxymanager database directly. for anyone interested here my changes:

replace line 64 in file [add-npm-geofilter.py](https://github.com/lavinir/npm-geoaccesslists/blob/6a9341f36d8d1928397b2bdb69c1011ab27fd10c/add-npm-geofilter.py#L64):
`npmclient.add_access_list(args.npm_accesslist_name, token, existing_rule_id, *access_rule_list)`

with this code:

```
import sqlite3
import datetime

connection = sqlite3.connect("/path/to/nginxproxymanager/database.sqlite")
cursor = connection.cursor()

# remove existing networks of rule list
cursor.execute(f"DELETE FROM access_list_client WHERE access_list_id = {existing_rule_id}")

# add each entry to the DB
for row in access_rule_list:
    query_value_meta = '{}'
    cursor.execute(f"INSERT INTO access_list_client (created_on, modified_on, access_list_id, address, directive, meta) VALUES ('{datetime.datetime.now()}', '{datetime.datetime.now()}', {existing_rule_id}, '{row.address}', '{row.directive}', '{query_value_meta}')")

connection.commit()
print("synced to DB")
connection.close()
```
(replace "/path/to/nginxproxymanager/database.sqlite" with the path to your nginxproxymanager sqlite database)

**Comment by pw-h** [comment]
EDIT: never mind the remark below, the IP ranges appeared in the access list after 5 minutes, must have been that damned Chrome cache
EDIT2: adding 12416 entries to the access list made editing the access list from the gui impossible, constant time-outs. Luckily i added a test-accesslist which i could delete from the gui. So unfortunately the fix doesn't work for me.

Above fix doesn't seem to be working anymore, 

_# python add-npm-geofilter.py --npm-host 10.0.0.80 --npm-port 881 --npm-email xxxxx@xxxxxxxx.xxx --npm-password xxxxxxxxxxx --npm-accesslist-name Toegangslijst2 --allowed-countries NL
Fetching ip list from db-ip.com
Downloading https://download.db-ip.com/free/dbip-country-lite-2025-01.csv.gz to dbip-country-lite-2025-01.csv.gz
Found 12416 entries for ['NL']
Getting auth token
Adding access list
synced to DB_

But no new entries are visible in the "Access" tab of the Access list.  NPM v2.12.2 © 2024 [jc21.com]
